### DISCUSSION TITLE
Split services between multiple Pi's

### DISCUSSION THREAD

**Discussion by jgandy** [mcguirepr89/BirdNET-Pi#1031] [General]
Has anyone split the services across multiple Raspberry Pi devices to share the workload?  I've had it running on an older pi 3 b+, but now that Pi's are back in stock, I picked up a 4gb pi 4.  I want to leave the pi 3 outside doing the sound capture and move the work onto the pi 4.  Idea is to do install on secondary device, and just start the services required on each device.  Write files in nfs on NAS or pi4, have db on pi4.
I've searched and didn't find much.  There's a post about putting it in docker, and I saw mention of BirdCage.  Anyway, I'll post any updates here as i get into it.

services=(chart_viewer.service --> pi4
  spectrogram_viewer.service --> pi4
  icecast2.service --> is this the live stream? 
  extraction.service --> pi4 ?
  birdnet_recording.service --> pi3
  birdnet_log.service --> pi4
  birdnet_stats.service)--> pi4

**Comment by netnutmike** [comment]
I want to do something similar.  I have 2 PI's running Birdpi which means 2 different web itnerfaces.  They are too far apart to do any creative sound merging with mixers, etc.  Plus, they are different sides of the house and pickup different things.  My inital thought was to use something like rsync or maybe something more secure :).

I have a large K3S cluster running on Pi's and orangePis that could do the processing as well.  Your idea of only running services and using NFS makes a lot of sense to run the processing and web interface on the EKS cluster.

I have reversed engineered Thsi in the past but did not document it so I would have to relearn what the different processes are.

You mentioned docker, has anyone tried to dockerize any of the processes?  That would even make sense for it running on a single PI, easy to install and easy to manage.
